share/functions/fish_vi_cursor.fish

```diff
@@ -86,7 +32,7 @@ function fish_vi_cursor -d 'Set cursor shape for different vi modes'
               if not set -q \$varname
                 set varname fish_cursor_unknown
               end
-              $function \$\$varname
+              __fish_cursor_xterm \$\$varname
```

Comment:
LGTM, the reason why I didn't remove these checks is simply that I didn't look into what the cursor control sequences look like.
Looks like they use CSI (and have a space in the middle, oddly) which we already send.